src/OpenTK.GraphicsLibraryFramework/GLFWProvider.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace OpenToolkit.GraphicsLibraryFramework
+{
+    public static class GLFWProvider
+    {
+        private static Lazy<GLFW> glfw = null;
+        public static GLFW GLFW
```

Comment:
This isn't really a best-practice implementation of `Lazy`.

May I suggest this:
```cs
public static Lazy<GLFW> GLFW { get; } = new Lazy<GLFW>(() =>
{
    var glfw = GLFW.GetAPI();
    lock (glfw)
    {
        glfw.Init();
        glfw.SetErrorCallback(GLFW.errorCallback);
    }
});
```
Then all you need to do is replace `GLFWProvider.GLFW` with `GLFWProvider.GLFW.Value`. This is probably the best practice for Lazy implementation.

Reply:
Lazy without parameters uses the PublicationAndExecution thread safe mode, which is this part:
https://referencesource.microsoft.com/#mscorlib/system/Lazy.cs,377
So locking should not be necessary, as it is already done with a Monitor...

Reply:
Updated my suggestion.

Reply:
@jvbsl Thanks. Your insight and attentiveness is always appreciated on things like this.

Reply:
Fixed.